cli/init.js

```diff
@@ -18,11 +18,18 @@ module.exports = function init(args) {
   const relativePublicDir = path.relative(CWD, absolutePublicDir)
   const dirExists = fs.existsSync(absolutePublicDir)
 
-  invariant(
-    dirExists,
-    'Failed to create a Service Worker at "%s": directory does not exist.\nMake sure to include a relative path to the root directory of your server.',
-    absolutePublicDir,
-  )
+  if (!dirExists) {
+    // Try to create the directory if it doesn't exist
+    try {
+      fs.mkdirSync(absolutePublicDir, { recursive: true })
+    } catch {
+      invariant(
+        false,
+        'Failed to create a Service Worker at "%s": directory does not exist and could not be created.\nMake sure to include a relative path to the root directory of your server.',
```

Comment:
I think we should include the original error message in our custom one to contain more information when debugging any related issues.

```js
invariant(..., `\
Failed to create ...

See the original error below:
${error}\
`)
```